metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/bolt/ParserBolt.java

```diff
@@ -76,12 +77,15 @@ public void prepare(Map stormConf, TopologyContext context, OutputCollector coll
     if(getSensorParserConfig() == null) {
       filter = new GenericMessageFilter();
     }
-    else if(filter == null) {
+    if(filter == null) {
       getSensorParserConfig().getParserConfig().putIfAbsent("stellarContext", stellarContext);
-      filter = Filters.get(getSensorParserConfig().getFilterClassName()
-              , getSensorParserConfig().getParserConfig()
-      );
+      if(!StringUtils.isEmpty(getSensorParserConfig().getFilterClassName())) {
```

Comment:
`GenericMessageFilter` just returns true all the time.  Instead, we just check to see if filter is null in the loop and pass things by if it's null.  Removes an unnecessary function call.  If you think it's more clear to set filter to `GenericMessageFilter` if it's not otherwise specified, I can do that.